src/Compilers/CSharp/Portable/Binder/Binder_InterpolatedString.cs

```diff
@@ -205,26 +203,264 @@ bool tryBindAsHandlerType([NotNullWhen(true)] out BoundInterpolatedString? resul
             }
         }
 
+        private static bool AllInterpolatedStringPartsAreStrings(ImmutableArray<BoundExpression> parts)
+            => parts.All(p => p is BoundLiteral or BoundStringInsert { Value: { Type: { SpecialType: SpecialType.System_String } }, Alignment: null, Format: null });
+
+        private bool TryBindUnconvertedBinaryOperatorToDefaultInterpolatedStringHandler(BoundBinaryOperator binaryOperator, BindingDiagnosticBag diagnostics, [NotNullWhen(true)] out BoundBinaryOperator? convertedBinaryOperator)
+        {
+            // Much like BindUnconvertedInterpolatedStringToString above, we only want to use DefaultInterpolatedStringHandler if it's worth it. We therefore
+            // check for cases 1 and 2: if they are present, we let normal string binary operator binding machinery handle it. Otherwise, we take care of it ourselves.
+            Debug.Assert(binaryOperator.IsUnconvertedInterpolatedStringAddition);
+            convertedBinaryOperator = null;
+
+            var interpolatedStringHandlerType = Compilation.GetWellKnownType(WellKnownType.System_Runtime_CompilerServices_DefaultInterpolatedStringHandler);
+            if (interpolatedStringHandlerType.IsErrorType())
+            {
+                // Can't ever bind to the handler no matter what, so just let the default handling take care of it. Cases 4 and 5 are covered by this.
+                return false;
+            }
+
+            bool isConstant = true;
+            var stack = ArrayBuilder<BoundBinaryOperator>.GetInstance();
+            var partsArrayBuilder = ArrayBuilder<ImmutableArray<BoundExpression>>.GetInstance();
+            int partsCount = 0;
+
+            BoundBinaryOperator? current = binaryOperator;
+
+            while (current != null)
+            {
+                Debug.Assert(current.IsUnconvertedInterpolatedStringAddition);
+                stack.Push(current);
+                isConstant = isConstant && current.Right.ConstantValue is not null;
```

Comment:
nit: `&&=`?

Reply:
This is the second or third time you've asked for this operator. Perhaps you should consider seeing if LDM wants to reconsider the `Likely Never` milestone on [https://github.com/dotnet/csharplang/issues/1718](https://github.com/dotnet/csharplang/issues/1718) 😉

Reply:
Oops. This one keeps tripping me up. I even thought I'd checked on sharplab yesterday. :-P